src/codecs/tiff.rs

```diff
@@ -296,24 +381,91 @@ fn cmyk_to_rgb(cmyk: &[u8]) -> [u8; 3] {
     ]
 }
 
-// Utility to simplify and deduplicate error handling during 16-bit encoding.
-fn u8_slice_as_u16(buf: &[u8]) -> ImageResult<&[u16]> {
-    bytemuck::try_cast_slice(buf).map_err(|err| {
-        // If the buffer is not aligned or the correct length for a u16 slice, err.
-        //
-        // `bytemuck::PodCastError` of bytemuck-1.2.0 does not implement
-        // `Error` and `Display` trait.
-        // See <https://github.com/Lokathor/bytemuck/issues/22>.
-        ImageError::Parameter(ParameterError::from_kind(ParameterErrorKind::Generic(
-            format!("{:?}", err),
-        )))
-    })
+enum DtypeContainer<'a, T> {
+    Slice(&'a [T]),
+    Vec(Vec<T>),
+}
```

Comment:
This looks much like `std::borrow::Cow`. What's the difference for not using that standard type?